src/main/java/sparksoniq/jsoniq/runtime/iterator/primary/VariableReferenceIterator.java

```diff
@@ -44,19 +45,12 @@
     private List<Item> _items = null;
     private int _currentIndex = 0;
 
-    public VariableReferenceIterator(String variableName, SequenceType seq, IteratorMetadata iteratorMetadata) {
-        super(null, iteratorMetadata);
-        _variableName = "$" + variableName;
+    public VariableReferenceIterator(String variableName, SequenceType seq, ExecutionMode executionMode, IteratorMetadata iteratorMetadata) {
+        super(null, executionMode, iteratorMetadata);
+        _variableName = variableName;
```

Comment:
In order to be consistent across the application, I have standardized the variable names in wherever they are used. Variables are now stored and referenced in the dynamic context without the $ prefix. They are also mentioned in error messages without the prefix. I have opted for this as the behavior now matches Zorba.

Reply:
Outdated